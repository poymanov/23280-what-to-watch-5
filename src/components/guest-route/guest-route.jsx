import React, {PureComponent} from "react";
import PropTypes from "prop-types";
import {Route, Redirect} from "react-router-dom";
import {connect} from "react-redux";
import {AppRoute} from "../../constants/const";
import {isUserAuthSelector} from "../../store/selectors";
import {checkAuth} from "../../store/api-actions";

class GuestRoute extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.checkAuth();
  }

  render() {
    const {render, path, exact, isUserAuth} = this.props;

    return (
      <Route
        path={path}
        exact={exact}
        render={(routeProps) => {
          return (
            isUserAuth ? <Redirect to={AppRoute.ROOT}/> : render(routeProps)
          );
        }}
      />
    );
  }
}


GuestRoute.propTypes = {
  isUserAuth: PropTypes.bool.isRequired,
  exact: PropTypes.bool.isRequired,
  path: PropTypes.string.isRequired,
  render: PropTypes.func.isRequired,
  checkAuth: PropTypes.func.isRequired,
};

const mapStateToProps = (state) => ({
  isUserAuth: isUserAuthSelector(state),
});

const mapDispatchToProps = (dispatch) => ({
  checkAuth() {
    dispatch(checkAuth());
  },
});

export {GuestRoute};
export default connect(mapStateToProps, mapDispatchToProps)(GuestRoute);
