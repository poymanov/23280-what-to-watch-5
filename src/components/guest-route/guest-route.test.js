import React from "react";
import renderer from "react-test-renderer";
import {GuestRoute} from "./guest-route";
import {Router as BrowserRouter} from "react-router-dom";
import browserHistory from "../../etc/browser-history";

const noop = () => {};

it(`Should GuestRoute render correctly`, () => {
  const tree = renderer
    .create(<BrowserRouter history={browserHistory}>
      <GuestRoute isUserAuth={false} authorizationStatus={``} exact={true} path={`/`} render={noop} checkAuth={noop} />
    </BrowserRouter>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
