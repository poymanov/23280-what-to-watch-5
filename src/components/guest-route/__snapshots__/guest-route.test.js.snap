// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should GuestRoute render correctly 1`] = `null`;
